### PR TITLE
add support for hidePostCTA setting

### PR DESCRIPTION
**Pull request by reytum**
* If the setting is enabled, all the rich messages after the last sent message will be visible
* The buttons not having any type in card/list will not be affected since the check is only based on the type
* In web currently only quick replies are hidden, I have provided setting for hiding all three types, links, submit and quickReply.

### CODE REVIEW

**Review by devashishmamgain — CHANGES_REQUESTED** [review]

**Comment by devashishmamgain** [comment]
@reytum can you share apk also

File: kommunicateui/src/main/java/com/applozic/mobicomkit/uiwidgets/conversation/fragment/MobiComConversationFragment.java [hunk 1]
```diff
@@ -4750,4 +4765,9 @@ protected void toggleMessageSendLayoutVisibility() {
             individualMessageSendLayout.setVisibility(VISIBLE);
         }
     }
+
+    protected void checkMessageIndex(Message message) {
+        int index = messageList.indexOf(message);
```
[the remark this answers is not in the document]
Comment (reytum) [reply kommunicateui/src/main/java/com/applozic/mobicomkit/uiwidgets/conversation/fragment/MobiComConversationFragment.java:4770]: Sorry forgot to remove it.

File: kommunicate/src/main/java/com/applozic/mobicomkit/api/conversation/Message.java [hunk 2]
```diff
@@ -717,6 +717,10 @@ public boolean isIgnoreMessageAdding(Context context) {
         return ((ApplozicClient.getInstance(context).isActionMessagesHidden() && isActionMessage()) || hasHideKey());
     }
 
+    public boolean isRichMessage() {
+        return metadata != null && "300".equals(metadata.get("contentType"));
```
[the remark this answers is not in the document]
Comment (reytum) [reply kommunicate/src/main/java/com/applozic/mobicomkit/api/conversation/Message.java:721]: Ok

**Review by devashishmamgain — APPROVED** [review]

**Review by devashishmamgain — CHANGES_REQUESTED** [review]
generic card, list template, images, etc is not working

**Comment by reytum** [comment]
> generic card, list template, images, etc is not working

Did the changes. Now all the rich messages after the last sent message will be visible.

**Review by devashishmamgain — APPROVED** [review]

**Pull request merged**